app/src/test/java/com/codesoom/assignment/application/TaskServiceTest.java

```diff
@@ -0,0 +1,171 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.TaskNotFoundException;
+import com.codesoom.assignment.models.Task;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@DisplayName("TaskService 클래스")
+class TaskServiceTest {
+    private TaskService taskService;
+
+    private static final long VALID_TEST_ID = 1L;
+    private static final long INVALID_TEST_ID = 20000L;
+    private static final String TEST_TITLE = "testTitle";
+    private static final String PARAM_TEST_TITLE = "paramTestTitle";
+
+    //질문 Java에서 익명클래스의 장황함을 개선하기 위해 람다를 사용하는 이유와 같은 느낌의 장황함을 느낌.
+    @BeforeEach
+    //fixtures
+    void setUp() {
+        //subject : 테스트하는 대상
+        taskService = new TaskService();
+
+        Task task = new Task();
+        task.setTitle(TEST_TITLE);
+        taskService.createTask(task);
+    }
+
+    private Task makeSource(){
+        Task task = new Task();
+        task.setTitle(PARAM_TEST_TITLE);
+        return task;
+    }
+
+    @Nested
+    @DisplayName("list 메소드는")
+    class Describe_list{
+        @Nested
+        @DisplayName("만약 클라이언트 호출을 받으면")
+        class Context_wiht_an_clinent_call{
+            @Test
+            @DisplayName("list를 리턴한다.")
+            void getTasks() {
+                assertThat(taskService.getTasks()).isInstanceOf(List.class);
+            }
+        }
+    }
+
+
+    @Nested
+    @DisplayName("get 메소드는")
+    class Describe_get{
+        @Nested
+        @DisplayName("만약 올바른 id를 전달받았다면")
+        class Context_with_VALID_ID_id {
+            @Test
+            @DisplayName("id에 해당하는 task를 리턴한다.")
+            void it_returns_a_task(){
+                Task result = taskService.getTask(VALID_TEST_ID);
+
+                assertThat(result.getTitle()).isEqualTo(TEST_TITLE);
+                assertThat(result).isInstanceOf(Task.class);
+            }
+        }
+
+        @Nested
+        @DisplayName("만약 올바르지 못한 id를 전달받았다면")
+        class Context_with_invalid_id{
+            @Test
+            @DisplayName("Exception을 발생한다.")
+            void it_returns_an_exception(){
+                //when(taskService.getTask(INVALID_ID)).thenThrow(TaskNotFoundException.class);
+                assertThatThrownBy(() -> taskService.getTask(INVALID_TEST_ID))
+                        .isInstanceOf(TaskNotFoundException.class);
+            }
```

Comment:
예외를 다룰 때 사용하는 키워드들을 떠올려 보세요. `throw`, `throws`, `@throw`, `assertThatThrownBy()`... 전부 "던진다"는 표현을 쓰고 있죠? 남이 만든 프로그램을 사용할 때에는 예외가 마치 "발생하는 것"처럼 보일 거에요. 하지만 코드를 직접 작성하는 개발자의 입장이라면 예외는 발생하는 것과는 거리가 있습니다. 내가 작성한 코드가 예외를 던지는 거죠.

두 문장을 비교해 여러 차례 소리내어서 읽어보세요. 첫 번째 문장은 동호님이 작성한 걸 이어붙인 거고, 두 번째 문장은 제가 작성해봤습니다.

- get 메소드는, (만약 올바르지 못한 id를 전달받았다면,) Exception을 발생한다.
- get 메소드는, (등록되지 않은 할 일의 id가 주어지면,) 할 일을 찾을 수 없다는 예외를 던진다.

```suggestion
            @DisplayName("할 일을 찾을 수 없다는 예외를 던진다.")
            void it_returns_an_exception(){
                //when(taskService.getTask(INVALID_ID)).thenThrow(TaskNotFoundException.class);
                assertThatThrownBy(() -> taskService.getTask(INVALID_TEST_ID))
                        .isInstanceOf(TaskNotFoundException.class);
            }
```